### PR TITLE
feat: Add filter flag to allow generating configs for subset of folders

### PR DESCRIPTION
**Pull request by GenPage**
# Pull Request

## Related Github Issues

- runatlantis/atlantis#1466

## Description

I was originally using the root flag for this but realized this caused
unintended issues with Atlantis and how the pattern matching is done.

This provides an additional flag "--filter" that is passed to
FindConfigFilesInPath() to only search a sub-path based on root.

## Security Implications

- None

## System Availability

- All

### CODE REVIEW

**Comment by GenPage** [comment]
Looks like the feature will need a bit more work after adding tests. They are currently failing with:

```
=== RUN   TestFilterFlagWithInfraLiveProd
time="2021-03-19T12:22:36-04:00" level=info msg="Could not find an old config file. Starting from scratch"
Error: Rel: can't make /Users/paged/workspace/terragrunt-atlantis-config/test_examples/terragrunt-infrastructure-live-example/prod/account.hcl relative to ../test_examples/terragrunt-infrastructure-live-example/prod/us-east-1/prod/mysql/

    generate_test.go:62: open test_artifacts/8563953357423524462.yaml: no such file or directory
--- FAIL: TestFilterFlagWithInfraLiveProd (0.01s)
=== RUN   TestFilterFlagWithInfraLiveNonProd
time="2021-03-19T12:22:36-04:00" level=info msg="Could not find an old config file. Starting from scratch"
Error: Rel: can't make /Users/paged/workspace/terragrunt-atlantis-config/test_examples/terragrunt-infrastructure-live-example/non-prod/account.hcl relative to ../test_examples/terragrunt-infrastructure-live-example/non-prod/us-east-1/qa/mysql/

    generate_test.go:62: open test_artifacts/3173967700786741497.yaml: no such file or directory
--- FAIL: TestFilterFlagWithInfraLiveNonProd (0.01s)
FAIL
FAIL    github.com/transcend-io/terragrunt-atlantis-config/cmd  0.903s
FAIL
make: *** [test] Error 1
```

File: cmd/generate.go [hunk 1]
```diff
@@ -316,7 +316,15 @@ func getAllTerragruntFiles() ([]string, error) {
 		return nil, err
 	}
 
-	paths, err := config.FindConfigFilesInPath(gitRoot, options)
+	// If filterPath is provided, override workingPath instead of gitRoot
+	// We do this here because we want to keep the relative path structure of Terragrunt files
+	// to root and just ignore the ConfigFiles
+	workingPath := gitRoot
+	if filterPath != "" {
+		workingPath = filterPath
```
Comment (dmattia) [inline cmd/generate.go:324]:
```suggestion
		workingPath, err = filepath.Abs(filterPath)
		if err != nil {
			return nil, err
		}
```

**Review by dmattia — CHANGES_REQUESTED** [review]
The two new tests are failing, which I believe to be because some of these functions require absolute paths passed in, while the `filterPath` specified in the tests is relative.

I added a suggestion for where I believe the `filepath.Abs` call ought to go, please let me know if you have other suggestions.

Thank you very much for this wonderful PR with great comments and tests, I will merge and release as soon as tests pass

**Comment by GenPage** [comment]
@dmattia Ah! Make sense! I don't have any other suggestions at this time. The tests pass locally with your suggestion.

Thank you for a wonderful tool. I appreciate your kind words. :)

**Review by dmattia — APPROVED** [review]

**Comment by dmattia** [comment]
Released in https://github.com/transcend-io/terragrunt-atlantis-config/releases/tag/v1.2.1 @GenPage

**Pull request merged**